src/vs/workbench/contrib/notebook/browser/notebook.contribution.ts

```diff
@@ -285,223 +239,16 @@ export class NotebookContribution extends Disposable implements IWorkbenchContri
-			return this._handleDiffEditorInput(originalInput, options, group);
-		}
-
-		if (!originalInput.resource) {
-			return undefined;
-		}
-
-		// Run reopen with ...
-		if (id) {
-			// from the editor tab context menu
-			if (originalInput instanceof NotebookEditorInput) {
-				if (originalInput.viewType === id) {
-					// reopen with the same type
-					return undefined;
-				} else {
-					return {
-						override: (async () => {
-							const notebookInput = NotebookEditorInput.create(this.instantiationService, originalInput.resource, id);
-							const originalEditorIndex = group.getIndexOfEditor(originalInput);
-
-							await group.closeEditor(originalInput);
-							originalInput.dispose();
-							const newEditor = await group.openEditor(notebookInput, { ...options, index: originalEditorIndex, override: EditorOverride.DISABLED });
-							if (newEditor) {
-								return newEditor;
-							} else {
-								return undefined;
-							}
-						})()
-					};
-				}
-			} else {
-				// from the file explorer
-				const existingEditors = this.editorService.findEditors(originalInput.resource, group).filter(editor => editor instanceof NotebookEditorInput) as NotebookEditorInput[];
-
-				if (existingEditors.length) {
-					// there are notebook editors with the same resource
-
-					if (existingEditors.find(editor => editor.viewType === id)) {
-						return { override: this.editorService.openEditor(existingEditors.find(editor => editor.viewType === id)!, { ...options, override: EditorOverride.DISABLED }, group) };
-					} else {
-						return {
-							override: (async () => {
-								const firstEditor = existingEditors[0]!;
-								const originalEditorIndex = group.getIndexOfEditor(firstEditor);
-
-								await group.closeEditor(firstEditor);
-								firstEditor.dispose();
-								const notebookInput = NotebookEditorInput.create(this.instantiationService, originalInput.resource!, id);
-								const newEditor = await group.openEditor(notebookInput, { ...options, index: originalEditorIndex, override: EditorOverride.DISABLED });
-
-								if (newEditor) {
-									return newEditor;
-								} else {
-									return undefined;
-								}
-							})()
-						};
-					}
-				}
-			}
-		}
-
-		// Click on the editor tab
-		if (id === undefined && originalInput instanceof NotebookEditorInput) {
-			const existingEditors = this.editorService.findEditors(originalInput.resource, group).filter(editor => editor instanceof NotebookEditorInput && editor === originalInput);
-
-			if (existingEditors.length) {
-				// same
-				return undefined;
-			}
-		}
-
-		if (originalInput instanceof NotebookDiffEditorInput) {
-			return undefined;
-		}
-
-		let notebookUri: URI = originalInput.resource;
-		let cellOptions: IResourceEditorInput | undefined;
-
-		const data = CellUri.parse(originalInput.resource);
-		if (data) {
-			notebookUri = data.notebook;
-			cellOptions = { resource: originalInput.resource, options };
-		}
```

Comment:
I don't see where this code has gone. This isn't strictly about opening an editor for its file but about opening a notebook editor when there is a request to open one of its cell documents. You can test/verify this with github issue notebooks:

* have two cells, the first defines a variable, the second uses the variable (as in `.vscode/notebooks/api.github-issues`)
* in the second cell invoke go-to-definition on the variable
* make sure you land in the first cell, not in a separate text editor

This use-case is the reason I was lobbying for the supporting registrations per schemes - all notebook cells use the `vscode-notebook-cell` scheme

Reply:
This case still works, @rebornix helped me with it. We actually don't need the scheme registration and I plan to remove it as what we do is match the glob pattern of `*.github-issue` to the cell uri and then we pass you the cell uri + the viewType of `github-issue` and the notebook codes resolves that cell uri and returns an `IEditorInputWithOptions`

Reply:
Ah, now I see where that went. OK, so you are relying on the fact that cell uris inherit the original path. Maybe a little more magic/implicit than I would have liked

Reply:
I don't think it's possible to support the scheme registration though. How would you know what viewtype the scheme should be? The scheme needs to map to a specific viewpoint.

Reply:
I see. I would support all view-types for that scheme and figure out the actual view-type later. Not very pretty. Let's stick with this solution and take a note that we don't ever change the way cell uri are constructed...